### ISSUE TITLE
CRC Image Registry login Issue - dial tcp 192.168.130.11:5000: connect: connection refused

### ISSUE THREAD

**Issue by hi2kiran** [code-ready/crc#1146]
### General information

  * OS: Ubuntu 18.04.4 LTS (Bionic Beaver)" VM with 64 GB RAM, 24 vCPUs, 100 GB disk
  * Hypervisor: KVM
  * Did you run `crc setup` before starting it (Yes/No)? Yes

$ crc version
crc version: 1.7.0+fa7e558
OpenShift version: 4.3.1 (embedded in binary)

$ crc status
CRC VM:          Running
OpenShift:       Running (v4.3.1)
Disk Usage:      16.72GB of 32.72GB (Inside the CRC VM)
Cache Usage:     12.3GB
Cache Directory: /home/ubuntu/.crc/cache

## CRC config
$ crc config view
- skip-check-crc-dnsmasq-file           : true
- skip-check-network-manager-config     : true
- skip-check-network-manager-installed  : true
- skip-check-network-manager-running    : true

## Host Operating System

NAME="Ubuntu"
VERSION="16.04.6 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.6 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial 

### Steps to reproduce

  1. crc setup
  2. crc start
     oc login -u kubeadmin -p <passwd> https://api.crc.testing:6443
  3. Followed the instructions mentioned at https://docs.openshift.com/container-platform/4.3/registry/accessing-the-registry.html
  $ oc policy add-role-to-user registry-viewer developer
	clusterrole.rbac.authorization.k8s.io/registry-viewer added: "developer"
  $ oc policy add-role-to-user registry-viewer kubeadmin
	clusterrole.rbac.authorization.k8s.io/registry-viewer added: "kubeadmin"
  $ oc policy add-role-to-user registry-editor developer
	clusterrole.rbac.authorization.k8s.io/registry-editor added: "developer"
  $ oc policy add-role-to-user registry-editor kubeadmin
	clusterrole.rbac.authorization.k8s.io/registry-editor added: "kubeadmin"
  4. podman login -u kubeadmin -p $(oc whoami -t) image-registry.openshift-image-registry.svc:5000

### Expected
Login Succeeded!

### Actual

$ podman login -u kubeadmin -p $(oc whoami -t) image-registry.openshift-image-registry.svc:5000
WARNING! Using --password via the cli is insecure. Please consider using --password-stdin
Error: error authenticating creds for "image-registry.openshift-image-registry.svc:5000": error pinging docker registry image-registry.openshift-image-registry.svc:5000: Get https://image-registry.openshift-image-registry.svc:5000/v2/: dial tcp 192.168.130.11:5000: connect: connection refused

### Logs

$ curl -vk https://image-registry.openshift-image-registry.svc:5000/v2/
*   Trying 192.168.130.11...
* TCP_NODELAY set
* connect to 192.168.130.11 port 5000 failed: Connection refused
* Failed to connect to image-registry.openshift-image-registry.svc port 5000: Connection refused
* Closing connection 0
curl: (7) Failed to connect to image-registry.openshift-image-registry.svc port 5000: Connection refused

$ oc project openshift-image-registry
Now using project "openshift-image-registry" on server "https://api.crc.testing:6443".

$ oc get pods
NAME                                              READY   STATUS    RESTARTS   AGE
cluster-image-registry-operator-79bfcb686-4g44v   2/2     Running   0          31d
image-registry-5f4d9dc667-dhmxz                   1/1     Running   0          31d
node-ca-9s7kb                                     1/1     Running   0          31d

$ og c get all
NAME                                                  READY   STATUS    RESTARTS   AGE
pod/cluster-image-registry-operator-79bfcb686-4g44v   2/2     Running   0          31d
pod/image-registry-5f4d9dc667-dhmxz                   1/1     Running   0          31d
pod/node-ca-9s7kb                                     1/1     Running   0          31d

NAME                              TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)     AGE
service/image-registry            ClusterIP   172.30.110.13   <none>        5000/TCP    31d
service/image-registry-operator   ClusterIP   None            <none>        60000/TCP   31d

NAME                     DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE
daemonset.apps/node-ca   1         1         1       1            1           kubernetes.io/os=linux   31d

NAME                                              READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/cluster-image-registry-operator   1/1     1            1           31d
deployment.apps/image-registry                    1/1     1            1           31d

NAME                                                        DESIRED   CURRENT   READY   AGE
replicaset.apps/cluster-image-registry-operator-79bfcb686   1         1         1       31d
replicaset.apps/image-registry-5bd685c64                    0         0         0       31d
replicaset.apps/image-registry-5f4d9dc667                   1         1         1       31d
replicaset.apps/image-registry-66997968c6                   0         0         0       31d
replicaset.apps/image-registry-7dbbbf5479                   0         0         0       31d

NAME                                     HOST/PORT                                                 PATH   SERVICES         PORT    TERMINATION   WILDCARD
route.route.openshift.io/default-route   default-route-openshift-image-registry.apps-crc.testing          image-registry   <all>   reencrypt     None

**Comment by praveenkumar** [comment]
> podman login -u kubeadmin -p $(oc whoami -t) image-registry.openshift-image-registry.svc:5000

@hi2kiran this is wrong, looks like you added the `svc` route to your dns so it is resolving the `192.168.130.11` but you should use the route instead of service endpoints to access the registry.

```
$  podman login -u kubeadmin -p $(oc whoami -t) default-route-openshift-image-registry.apps-crc.testing --tls-verify=false
```

**Comment by hi2kiran** [comment]
Thanks a lot Praveen. It woked.
$ podman login -u kubeadmin -p $(oc whoami -t) default-route-openshift-image-registry.apps-crc.testing --tls-verify=false
WARNING! Using --password via the cli is insecure. Please consider using --password-stdin
Login Succeeded!

I have another Generic issue. If i execute any oc/crc commands ( like oc get pods, oc login, crc status). It is taking almost 15 secs to respond. Is there anything to be configured?

**Comment by gbraad** [comment]
This is a separate issue. Can we file it as such to look into this?

Closing this issue as it is resolved.
